src/Chargeback.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+
+namespace Omnipay\BlueSnap;
+
+use Omnipay\Common\Helper;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+/**
+ * Generic representation of a chargeback object returned by a gateway.
+ * Hopefully this can be added to omnipay-common one day.
+ */
+class Chargeback
+{
+    /**
+     * @var ParameterBag
+     */
+    protected $parameters;
+
+    /**
+     * Create a new chargeback with the specified parameters
+     *
+     * @param array $parameters An array of parameters to set on the new object
+     */
+    public function __construct($parameters = array())
+    {
+        $this->initialize($parameters);
+    }
+
+    /**
+     * Initialize this chargeback with the specified parameters
+     *
+     * @param array $parameters An array of parameters to set on this object
+     * @return static
+     */
+    public function initialize($parameters = array())
+    {
+        $this->parameters = new ParameterBag;
+
+        Helper::initialize($this, $parameters);
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getParameters()
+    {
+        return $this->parameters->all();
+    }
+
+    /**
+     * @param string $key
+     * @return mixed
+     */
+    protected function getParameter($key)
+    {
+        return $this->parameters->get($key);
+    }
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     * @return Chargeback
+     */
+    protected function setParameter($key, $value)
+    {
+        $this->parameters->set($key, $value);
+
+        return $this;
+    }
+
+    /**
+     * Get the chargeback id (not supported by BlueSnap)
+     *
+     * @return null|string
+     */
+    public function getChargebackId()
+    {
+        return $this->getParameter('chargebackId');
+    }
+
+    /**
+     * Set the chargeback id (not supported by BlueSnap)
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setChargebackId($value)
+    {
+        return $this->setParameter('chargebackId', $value);
+    }
+
+    /**
+     * Get the chargeback reference
+     *
+     * @return null|string
+     */
+    public function getChargebackReference()
+    {
+        return $this->getParameter('chargebackReference');
```

Comment:
What's the difference between `chargebackId` and `chargebackReference`?

Reply:
Honestly not sure. I copied the refund and chargeback classes from the omnipay-vindicia repo in order to have a standard way of setting up these objects. They're not used in Vindica either so I'm not sure what the context behind this was. At some point I'd like to go back and standardize these more into a single repo but I'm going with how omnipay-vindicia handles this for now.

Reply:
Sounds good, definitely out of scope here but we should do it at some point